### PR TITLE
various fixes to RoomHeader

### PR DESCRIPTION
**Pull request by t3chguy**
+ don't show buttons if they don't do anything
+ don't show search when we're not in the room (it'll fail opaquely)
+ hide settings button when appropriate (it'd do nothing)

and some smaller fixes

closes vector-im/riot-web#3859

### CODE REVIEW

File: src/components/views/rooms/RoomHeader.js [hunk 1]
```diff
@@ -225,12 +226,17 @@ module.exports = React.createClass({
                 roomName = this.props.room.name;
             }
 
+            const innerName =
+                <EmojiText element="div"
+                           className={ "mx_RoomHeader_nametext " + (settingsHint ? "mx_RoomHeader_settingsHint" : "") }
+                           title={ roomName }>{roomName}</EmojiText>;
```
Comment (lukebarnard1) [inline src/components/views/rooms/RoomHeader.js]: A few things here:
 - props should be `{"someValue"}` (no spaces)
 - children should be `{ someChild }`

File: src/components/views/rooms/RoomHeader.js [hunk 2]
```diff
@@ -225,12 +226,17 @@ module.exports = React.createClass({
                 roomName = this.props.room.name;
             }
 
+            const innerName =
+                <EmojiText element="div"
+                           className={ "mx_RoomHeader_nametext " + (settingsHint ? "mx_RoomHeader_settingsHint" : "") }
```
Comment (lukebarnard1) [inline src/components/views/rooms/RoomHeader.js]: It'd be _really_ awesome if this used `classnames` which can be imported from `/src`. Then the indentation could be a bit nicer :grinning: (i.e. `someClasses = classnames({...})` ... `className={someClasses}` )

File: src/components/views/rooms/RoomHeader.js [hunk 3]
```diff
@@ -225,12 +226,17 @@ module.exports = React.createClass({
...
+                           title={ roomName }>{roomName}</EmojiText>;
+
+            if (this.props.onSettingsClick) {
```
Comment (lukebarnard1) [inline src/components/views/rooms/RoomHeader.js]: I'm pretty sure it's ok to pass `undefined` to `onClick`, so this is a bit redundant.

File: src/components/views/rooms/RoomHeader.js [hunk 4]
```diff
@@ -299,6 +305,14 @@ module.exports = React.createClass({
                 </AccessibleButton>;
         }
 
+        let search_button;
+        if (this.props.onSearchClick && this.props.inRoom) {
```
Comment (lukebarnard1) [inline src/components/views/rooms/RoomHeader.js]: Again I don't think we should be checking for the optional `onSearchClick`, let's allow `AccessibleButton` to decide whether to call `undefined` or not.

File: src/components/views/rooms/RoomHeader.js (hunk 3 above)
Comment (t3chguy) [inline src/components/views/rooms/RoomHeader.js]:
```
rageshake.js:61 Warning: Failed prop type: The prop `onClick` is marked as required in `AccessibleButton`, but its value is `undefined`.
in AccessibleButton (created by RoomHeader)
in RoomHeader (created by RoomView)
in div (created by RoomView)
in RoomView (created by LoggedInView)
in main (created by LoggedInView)
in div (created by LoggedInView)
in div (created by LoggedInView)
in LoggedInView (created by MatrixChat)
in MatrixChat
```

File: src/components/views/rooms/RoomHeader.js (hunk 1 above)
Comment (t3chguy) [inline src/components/views/rooms/RoomHeader.js]: Fair enough, I actually just copied that and did not change it. Will add a commit to remedy that

File: src/components/views/rooms/RoomHeader.js (hunk 3 above)
Comment (lukebarnard1) [inline src/components/views/rooms/RoomHeader.js]: But this isn't an `AccessibleButton`, it's a `div` so that error must be for the other thing.

**Comment by ara4n** [comment]
given the changes are approved, why isn't this merged? :)

**Pull request merged**